templates/trust-private-ca-on-all-nodes/daemonset.yaml

```diff
@@ -0,0 +1,72 @@
+{{- if .Values.global.privateCaCertsAddToHost.enabled }}
+################################
+## DaemonSet to mount the private root CA
+##
+## This can be used by enterprise with private
+## CAs that do not already install their root
+## certificate on the kubernetes nodes.
+#################################
+apiVersion: apps/v1
+kind: DaemonSet
+metadata:
+  name: {{ .Release.Name }}-private-ca
+  labels:
+    tier: platform
+    component: private-ca
+    release: {{ .Release.Name }}
+spec:
+  # allow update to occur all at once
+  updateStrategy:
+    type: RollingUpdate
+    rollingUpdate:
+      maxUnavailable: "100%"
+  selector:
+    matchLabels:
+      tier: platform
+      component: private-ca
+      release: {{ .Release.Name }}
+  template:
+    metadata:
+      labels:
+        tier: platform
+        component: private-ca
+        release: {{ .Release.Name }}
+    spec:
+      serviceAccountName: {{ .Release.Name }}-private-ca
+      containers:
+      - name: sleeper
+        image: ubuntu:20.04
+        command:
+          - "/bin/bash"
+          - "-c"
+        args:
+          - "ls -ltrah /private-ca-certs/* && cp -v /private-ca-certs/* /host-trust-store/ && sleep infinity"
+        imagePullPolicy: IfNotPresent
+        volumeMounts:
+        - name: hostcerts
+          mountPath: /host-trust-store
+        {{ range $secret_name := (.Values.global.privateCaCerts) }}
+        - name: {{ $secret_name }}
+          # mush be named as *.crt to be noticed by Docker
+          # after this is copied to docker trust store
+          mountPath: /private-ca-certs/{{ $secret_name }}.crt
+          subPath: cert.pem
+        {{- end }}
+        resources:
+          requests:
+            cpu: 1m
+            memory: "25Mi"
+          limits:
+            cpu: 50m
+            memory: "50Mi"
+      terminationGracePeriodSeconds: 1
+      volumes:
+      - name: hostcerts
+        hostPath:
+          path: {{ .Values.global.privateCaCertsAddToHost.hostDirectory }}/registry.{{ .Values.global.baseDomain }}/
```

Comment:
This path must already exist on the host to mount it. If the job of this container is to add certs to the host, I have a feeling the `/registry.{{ .Values.global.baseDomain }}/` part won't exist the firs time this runs. 

you could get around this by adding `type: DirectoryOrCreate` on the line below path. That should create an empty dir if it doesn't exist.

Reply:
I'm not sure if this is correct because it is working, and I would expect it to fail in the case that pre-existing is required since it is mounting the domain name directory. [Docker bind mounts](https://docs.docker.com/storage/bind-mounts/) create directories on the host if they do not exist.

Reply:
I went with the continuously refresh option. Also, I found unexpectedly that restartPolicy: Never is not allowed with daemonsets